internal/ansible/ansible.go

```diff
@@ -191,20 +203,21 @@ func (p Parameters) roleCmdFunc(ctx context.Context, path string) (cmdFuncType,
 		// gosec is disabled here because of G204. We should pay attention that user can't
 		// make command injection via command argument
 		return exec.CommandContext(ctx, p.RunnerBinary, append(cmdArgs, cmdOptions...)...) //nolint:gosec
-	}, nil
+	}
 }
 
 // GalaxyInstall Install non-exists collections/roles with ansible-galaxy cli
-func (p Parameters) GalaxyInstall(ctx context.Context, behaviorVars map[string]string, isCollectionRequirements, isRoleRequirements bool) error {
+func (p Parameters) GalaxyInstall(ctx context.Context, behaviorVars map[string]string, requirementsType string) error {
 	requirementsFilePath := runnerutil.GetFullPath(p.WorkingDirPath, galaxyutil.RequirementsFile)
 	var cmdArgs []string
 	var cmdOptions []string
-	if isCollectionRequirements {
+	switch requirementsType {
```

Comment:
I just created a new issue at https://github.com/multicloudlab/crossplane-provider-ansible/issues/81 as an enhancement in future release, as it looks here the requirements type are mutually exclusive for `roles` and `collections`. It seems Ansible requirements.yml natively support mixed references to both. Let me know if that makes sense or not?

Reply:
thanks